src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java

```diff
@@ -214,4 +216,65 @@ void deleteUserTest_shouldReturnErrorMessageAnd400_whenUserResourceMismatch() th
                 .andExpect(jsonPath("$.code", is(HttpStatus.BAD_REQUEST.value())))
                 .andDo(print());
     }
+
+    @Test
+    @WithMockUser(
+            username = "username-one",
+            password = "password-one",
+            authorities = {"ROLE_USER"})
+    void getUserDetailsByIdTest_shouldReturnUserDetailsAnd200Ok_whenIdIsValid() throws Exception {
+        // Given
+        Role role = dataHelper.createRole("ROLE_USER");
+        Country country = dataHelper.createCountry(1);
+        User user = dataHelper.createUser("one", role, country, null, 20);
+
+        String REQUEST_URL = URL + "/{userId}";
+
+        // When
+        MvcResult mvcResult = mockMvc.perform(get(REQUEST_URL, user.getId()).contentType(MediaType.APPLICATION_JSON))
+                // Then
+                .andExpect(status().isOk())
+                .andDo(print())
+                .andReturn();
+
+        String responseContent = mvcResult.getResponse().getContentAsString();
+        UserResponseDto responseDto = objectMapper.readValue(responseContent, new TypeReference<UserResponseDto>() {});
+
+        assertThat(responseDto).isNotNull();
+        assertAll(
+                "UserResponseDto assertions",
+                () -> assertThat(responseDto.getId()).isEqualTo(user.getId()),
+                () -> assertThat(responseDto.getUsername()).isEqualTo(user.getUsername()),
+                () -> assertThat(responseDto.getEmail()).isEqualTo(user.getEmail()),
+                () -> assertThat(responseDto.getFullName()).isEqualTo(user.getFullName()),
+                () -> assertThat(responseDto.getCountryId())
+                        .isEqualTo(user.getCountry().getId()),
+                () -> assertThat(responseDto.getAge()).isEqualTo(user.getAge()));
+    }
+
+    @Test
+    @WithMockUser(
+            username = "username-one",
+            password = "password-one",
+            authorities = {"ROLE_USER"})
+    void getUserDetailsByIdTest_shouldReturnErrorMessageAnd400_whenUserResourceMismatch() throws Exception {
+        // Given
+        Role role = dataHelper.createRole("ROLE_USER");
+        Country country = dataHelper.createCountry(1);
+        User user = dataHelper.createUser("one", role, country, null, 20);
+
+        long wrongUserId = user.getId() + 1;
+
+        String REQUEST_URL = URL + "/{userId}";
+
+        // When
+        mockMvc.perform(delete(REQUEST_URL, wrongUserId)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(user.getId())))
+                // Then
+                .andExpect(status().isBadRequest())
+                .andExpect(jsonPath("$.message", is(ErrorMessage.USER_RESOURCE_MISMATCH.getName())))
```

Comment:
There is no such exception in the service method:

![image](https://github.com/Healthy-Lifestyle-Assistant/backend/assets/93421769/3ab70459-ef73-4358-a77d-6dfd391529d0)